ouroboros-network/src/Ouroboros/Network/Chain.hs

```diff
@@ -123,7 +123,10 @@ validExtension
 validExtension c b = blockInvariant b
                   && headHash c == blockPrevHash b
                   && headSlot c <  blockSlot b
-                  && headBlockNo c == pred (blockNo b)
+                  -- TODO This is a hack which should be addressed through 'WithOrigin'
```

Comment:
@avieth when can we expect the sequel of #706? :slightly_smiling_face:  We need `WithOrigin` for `BlockNo` (in addition to `SlotNo`) too, as the genesis EBB has `BlockNo 0`, the empty chain cannot have `BlockNo 0` too.

If you're busy with the byron-proxy, I could take a stab at it instead.

Reply:
I did this, for ouroboros-network only https://github.com/input-output-hk/ouroboros-network/pull/774

It would probably be best if you did the follow up for ouroboros-consensus. I found there were some non-obvious choices to be made in there.